### PR TITLE
Update snarkVM to 0.5.3

### PR DESCRIPTION
**Pull request by ljedrz**

### CODE REVIEW

**Comment by ljedrz** [comment]
@howardwu the `snarkos-toolkit` and `WASM Tests` jobs will fail due to a feature mismatch; `toolkit` would compile if its `snarkvm-dpc` dependency didn't exclude `default-features`, but since changing it might have WASM implications, I'll leave that one to you, in case any new feature needs to be isolated.

File: parameters/examples/create_genesis_block.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ pub fn generate<C: BaseDPCComponents>() -> Result<Vec<u8>, TransactionError> {
         difficulty_target: 0x07FF_FFFF_FFFF_FFFF_u64,
         nonce: 0,
         pedersen_merkle_root_hash: PedersenMerkleRootHash([0u8; 32]),
-        proof: ProofOfSuccinctWork::default(),
+        proof: ProofOfSuccinctWork([0; 972]),
```
Comment (ljedrz) [inline parameters/examples/create_genesis_block.rs:53]: The `Default` impl is no longer available, so I just hardcoded the value it used to provide.

Comment (howardwu) [reply parameters/examples/create_genesis_block.rs:53]: This is correct, thank you for calling that out.

**Comment by howardwu** [comment]
@ljedrz can you file a PR in snarkVM with the feature flag change to `*/Cargo.toml`?

**Comment by ljedrz** [comment]
Rebased.

**Comment by ljedrz** [comment]
@howardwu how would you like the features to look like? I was able to compile `snarkos-toolkit` on its own by introducing the following new feature to `snarkvm-dpc`:
```
wasm = [
  "full",
  "snarkvm-algorithms/wasm",
  "snarkvm-gadgets/full",
  "snarkvm-parameters/remote",
]
```
and by adding `features = [ "wasm" ]` to `[dependencies.snarkvm-dpc]` in `snarkos-toolkit`, but I'm not sure if this is the way it should be done.

File: toolkit/Cargo.toml [hunk 2]
```diff
@@ -26,20 +26,16 @@ path = "benches/account.rs"
...
 features = [ "wasm" ]
 
 [dependencies.snarkvm-dpc]
-version = "0.4.0"
-default-features = false
-
-[dependencies.snarkvm-objects]
-version = "0.4.0"
+version = "0.5.1"
 default-features = false
```
Comment (howardwu) [inline toolkit/Cargo.toml:35]:
```suggestion
default-features = false
features = [ "wasm" ]
```

**Comment by howardwu** [comment]
@ljedrz can you update all snarkVM imports to 0.5.2?

**Comment by ljedrz** [comment]
I updated the TOML files, but while `snarkos-toolkit` was fixed, the WASM tests still failed; I found the cause and filed a fix in https://github.com/AleoHQ/snarkVM/pull/192.

**Comment by ljedrz** [comment]
Updated to snarkVM 0.3.

**Comment by ljedrz** [comment]
The tests are green @howardwu, the only failure is some newly activated `clippy` lints on nightly.

**Review by howardwu — APPROVED** [review]

**Pull request merged**